services/containers/local.go

```diff
@@ -109,8 +111,14 @@ func (l *local) ListStream(ctx context.Context, req *api.ListContainersRequest,
 }
 
 func (l *local) Create(ctx context.Context, req *api.CreateContainerRequest, _ ...grpc.CallOption) (*api.CreateContainerResponse, error) {
+	span := trace.SpanFromContext(ctx)
```

Comment:
Can we hook the beginning of gRPC methods globally, instead of having trace.SpanFromContext in many methods?

Reply:
You need to read the span from the context. We are creating spans for each of the methods as well.

Reply:
Can we reduce the boilerplate here by having a utility function that returns the current tracer from the ctx?

package internal

func Tracer(ctx context.Context) trace.Tracer {
   return  trace.SpanFromContext(ctx).Tracer()  
}

Then you don't have to repeat the boilerplate again and again.

Reply:
Each component that creates spans should have its own handle to its own `Tracer`, whether obtained from the global `TracerProvider` or one provided at initialization time.  The `Span#Tracer()` method has been removed upstream and shouldn't be relied on since that `Tracer` may contain invalid resource information or different exporter configuration than the current component expects.